### PR TITLE
adds a happy path test case for EncryptionStateController

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: pkg/operator/encryption/state_controller_encryption_test.go [hunk 1]
```diff
@@ -50,6 +55,49 @@ func TestEncryptionStateController(t *testing.T) {
...
+			destName:                 "encryption-config-kube-apiserver-test",
+			targetGRs: map[schema.GroupResource]bool{
+				schema.GroupResource{Group: "", Resource: "secrets"}: true,
+			},
+			initialResources: []runtime.Object{
+				createDummyKubeAPIPod("kube-apiserver-1", "kms"),
+				createSecretBuilder("kms", schema.GroupResource{"", "secrets"}, 1).
+					WithEncryptionKey([]byte("61def964fb967f5d7c44a2af8dab6865")).ToCoreV1Secret(),
+			},
+			// TODO: figure why base64 of 61def964fb967f5d7c44a2af8dab6865 doesn't match
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/state_controller_encryption_test.go]: this is interesting . . .

Comment (enj) [reply pkg/operator/encryption/state_controller_encryption_test.go]: Trailing newline:

```
$ echo 61def964fb967f5d7c44a2af8dab6865 | base64
NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjUK

$ echo -n 61def964fb967f5d7c44a2af8dab6865 | base64
NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjU=
```

File: pkg/operator/encryption/state_controller_encryption_test.go [hunk 2]
```diff
@@ -50,6 +55,49 @@ func TestEncryptionStateController(t *testing.T) {
...
+			//       the test expects NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjUK whereas the actual value is NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjU=
+			//expectedEncryptionCfg: createEncryptionCfgNoWriteKey("1", "NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjUK", "secrets"),
+			expectedEncryptionCfg: createEncryptionCfgNoWriteKey("1", "NjFkZWY5NjRmYjk2N2Y1ZDdjNDRhMmFmOGRhYjY4NjU=", "secrets"),
+
+			validateFunc: func(ts *testing.T, actions []clientgotesting.Action, destName string, expectedEncryptionCfg *apiserverconfigv1.EncryptionConfiguration) {
+				if len(actions) != 5 {
```
Comment (enj) [inline pkg/operator/encryption/state_controller_encryption_test.go]: We should probably validate actions a bit more specifically.

File: pkg/operator/encryption/test_helpers.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
+package encryption
```
[the remark this answers is not in the document]
Comment (p0lyn0mial) [reply pkg/operator/encryption/test_helpers.go]: yes, good idea, thx.

File: pkg/operator/encryption/state_controller_encryption_test.go (hunk 1 above)
Comment (p0lyn0mial) [reply pkg/operator/encryption/state_controller_encryption_test.go]: ah, you are right :)

File: pkg/operator/encryption/state_controller_encryption_test.go (hunk 2 above)
Comment (p0lyn0mial) [reply pkg/operator/encryption/state_controller_encryption_test.go]: we could validate the verbs, if that's not enough we could validate the verbs and the accompanying resources.

**Comment by p0lyn0mial** [comment]
@enj comments applied, thanks for the review, PTAL

File: pkg/operator/encryption/test_helpers.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+const (
+	encryptionSecretKeyDataForTest = "encryption.operator.openshift.io-key"
+)
+
+type secretBuilder struct {
```
[the remark this answers is not in the document]
Comment (p0lyn0mial) [reply pkg/operator/encryption/test_helpers.go]: okay, will remove in a follow-up PR.

File: pkg/operator/encryption/state_controller_encryption_test.go (hunk 2 above)
Comment (enj) [reply pkg/operator/encryption/state_controller_encryption_test.go]: I am thinking verb + resource string + resource name/ns always validated and optional full object validation.  Ideally we do this generically in every test case.

File: pkg/operator/encryption/test_helpers.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (enj) [reply pkg/operator/encryption/test_helpers.go]: I am okay either way.  Others might feel more strongly.

**Pull request merged**